frontend/catalyst/jax_tracer.py

```diff
@@ -406,24 +413,104 @@ def trace_observables(obs, qubit_states, p, num_wires, qreg):
         qubits = get_qubits_from_wires(wires, qubit_states, qreg)
         jax_obs = jprim.compbasis(*qubits)
     elif isinstance(obs, KNOWN_NAMED_OBS):
-        _, wires = op_args
-        qubits = get_qubits_from_wires(wires, qubit_states, qreg)
-        jax_obs = jprim.namedobs(type(obs).__name__, qubits[0])
+        jax_obs, qubits = named_obs(obs, op_args, qubit_states, qreg)
     elif isinstance(obs, qml.Hermitian):
-        matrix, wires = op_args
-        qubits = get_qubits_from_wires(wires, qubit_states, qreg)
-        jax_obs = jprim.hermitian(matrix, *qubits)
+        jax_obs, qubits = hermitian_obs(obs, op_args, qubit_states, qreg)
     elif isinstance(obs, qml.operation.Tensor):
         nested_obs = [trace_observables(o, qubit_states, p, num_wires, qreg)[0] for o in obs.obs]
         jax_obs = jprim.tensorobs(*nested_obs)
     elif isinstance(obs, qml.Hamiltonian):
         nested_obs = [trace_observables(o, qubit_states, p, num_wires, qreg)[0] for o in obs.ops]
         jax_obs = trace_hamiltonian(op_args, *nested_obs)
-    else:
+    elif obs._pauli_rep is not None:  # pylint: disable=protected-access
+        paulis = obs._pauli_rep  # pylint: disable=protected-access
+        jax_obs = pauli_sentence_to_rt_obs(paulis, qubit_states, qreg)
```

Comment:
From what I understand using the Pauli representation is a special case that would typically be more performant, but we should be able to handle the general case of a `Sum` or `Prod` appearing here as well.

Reply:
Yes, thanks! The more general use case is when these operators are mixed with `qml.Hermitian`. I didn't test the support in my first version :/ as didn't see them showing in the Lightning unit tests for new arith operators.

I've added the support now. However, it doesn't currently work with nested Hamiltonians as it's not supported in the runtime. I need to double-check how/if this is working in Lightning sims before taking the next action.

Reply:
Yes, thanks! The more general use case is when these operators are mixed with `qml.Hermitian`. 

I've added the support now. However, it doesn't currently work with nested Hamiltonians as it's not supported in the runtime. I need to double-check how/if this is working in Lightning sims before taking the next action.